addons/account/models/account_payment.py

```diff
@@ -403,12 +404,19 @@ def _compute_payment_method_line_fields(self):
             to_exclude = self._get_payment_method_codes_to_exclude()
             if to_exclude:
                 pay.available_payment_method_line_ids = pay.available_payment_method_line_ids.filtered(lambda x: x.code not in to_exclude)
-            if pay.payment_method_line_id.id not in pay.available_payment_method_line_ids.ids:
-                # In some cases, we could be linked to a payment method line that has been unlinked from the journal.
-                # In such cases, we want to show it on the payment.
-                pay.hide_payment_method_line = False
+
+    @api.depends('payment_type')
+    def _compute_available_journal_ids(self):
+        """
+        Get all journals having at least one payment method for inbound/outbound depending on the payment_type.
+        Done with filtered because the domain does not work as expected on these fields.
+        """
+        for pay in self:
+            journals = self.env['account.journal'].search([('type', 'in', ('bank', 'cash'))])
```

Comment:
Since it is for a non stored computed field only visible in the form view, this is acceptable. But a `search` in a loop is not a good idea in general performance-wise.

Also, you are missing the `company_id` in the domain; you didn't forget it on the `account.payment.register`... And this is also why [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). A solution for that would be to use a mixin (`models.AbstractModel`) inherited by both `account.payment` and `account.payment.register`.
This [was the case in version 12.0](https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_payment.py) but was removed here https://github.com/odoo/odoo/commit/3c2e3b8c5a7d073fe8f0f6d781b1ccdbc320edd0. However, I feel like this is not some complicated logic that will need to be weirdly inherited like it was before.
Does @qdp-odoo have an opinion on this?

Reply:
I guess we won't have QDP opinion on this :slightly_smiling_face: 

I've rebased and had to adapt some things following some changes in the payment register.
Also tried something to remove the search in the loop, I think it is better but please tell me!